packages/headless/src/features/analytics/analytics-utils.ts

```diff
@@ -117,6 +119,26 @@ export const partialDocumentInformation = (
       ({uniqueId}) => result.uniqueId === uniqueId
     ) || 0;
 
+  return buildPartialDocumentInformation(result, resultIndex, state);
+};
+
+export const partialRecommendationInformation = (
+  result: Result,
+  state: Partial<RecommendationAppState>
+): PartialDocumentInformation => {
+  const resultIndex =
+    state.recommendation?.recommendations.findIndex(
+      ({uniqueId}) => result.uniqueId === uniqueId
+    ) || 0;
```

Comment:
Question on this ternary, similar to the one on line 120:

If the `recommendation` state property does not exist, the index is `0`, and the document position is `1` (see line 149).
If the `recommendation` state exists, but the result is not found in the array, the index is `-1` and document position is `0`.

It does not make sense to me to send two different values for the same condition "result not found".
However, I'm not clear on what the correct value for `documentPosition` should be, `0` or `1`?

Reply:
* Those documents not being found * should * not happen, since it's not really logical to be able to click on a document that does not exist. But.. 🤷 

* It's there as a fallback/artefact of the type system being unable to assure us that a given value will really be in the array, (which is normal, I guess 🤔 ) Another approach would be to just throw an error and not log anything. Which should happen upstream anyway, not in this function, which is pretty deep in the codebase.

* For now I would suggest that `index` should be 0, and `documentPosition` -1 when it's not found. Which most likely means it's going to be rejected by UA (I would assume...).

In any case, I guess just keep in mind that no matter what, if we are unable to find the given result unique ID in the list of results, there's not much we can do to make the UA payload "correct", it's most likely going to fail in one manner or the other. And it being 0, or -1, is a small detail versus the fact that there's some code upstream that is completely wrong when calling this function.

Reply:
Alright, I don't feel this is important enough to solve right now. It's definitely niche, but I still think it would be good to addressed, whether through a value, or some kind of feedback to the devs. I've created a [jira to revisit](https://coveord.atlassian.net/browse/KIT-762).